src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,218 @@
+    public static final String INPUT_IMAGE = "inputImage";
+
+    @VisibleForTesting
+    private final String modelId;
+    private final String embedding;
+    private final Map<String, String> fieldMap;
+
+    private final MLCommonsClientAccessor mlCommonsClientAccessor;
+    private final Environment environment;
+
+    public TextImageEmbeddingProcessor(
+        String tag,
+        String description,
+        String modelId,
+        String embedding,
+        Map<String, String> fieldMap,
+        MLCommonsClientAccessor clientAccessor,
+        Environment environment
+    ) {
+        super(tag, description);
+        if (StringUtils.isBlank(modelId)) throw new IllegalArgumentException("model_id is null or empty, can not process it");
+        validateEmbeddingConfiguration(fieldMap);
+
+        this.modelId = modelId;
+        this.embedding = embedding;
+        this.fieldMap = fieldMap;
+        this.mlCommonsClientAccessor = clientAccessor;
+        this.environment = environment;
+    }
+
+    private void validateEmbeddingConfiguration(Map<String, String> fieldMap) {
+        if (fieldMap == null
+            || fieldMap.isEmpty()
+            || fieldMap.entrySet()
+                .stream()
+                .anyMatch(x -> StringUtils.isBlank(x.getKey()) || Objects.isNull(x.getValue()) || StringUtils.isBlank(x.getValue()))) {
+            throw new IllegalArgumentException("Unable to create the TextImageEmbedding processor as field_map has invalid key or value");
+        }
+
+        if (fieldMap.entrySet().stream().anyMatch(entry -> !Set.of(TEXT_FIELD_NAME, IMAGE_FIELD_NAME).contains(entry.getKey()))) {
+            throw new IllegalArgumentException("Unable to create the TextImageEmbedding processor as field_map has unsupported field name");
+        }
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        return ingestDocument;
+    }
+
+    /**
+     * This method will be invoked by PipelineService to make async inference and then delegate the handler to
+     * process the inference response or failure.
+     * @param ingestDocument {@link IngestDocument} which is the document passed to processor.
+     * @param handler {@link BiConsumer} which is the handler which can be used after the inference task is done.
+     */
+    @Override
+    public void execute(final IngestDocument ingestDocument, final BiConsumer<IngestDocument, Exception> handler) {
+        try {
+            validateEmbeddingFieldsValue(ingestDocument);
+            Map<String, String> knnMap = buildMapWithKnnKeyAndOriginalValue(ingestDocument);
+            Map<String, String> inferenceMap = createInferences(knnMap);
+            if (inferenceMap.isEmpty()) {
+                handler.accept(ingestDocument, null);
+            } else {
+                mlCommonsClientAccessor.inferenceSentences(this.modelId, inferenceMap, ActionListener.wrap(vectors -> {
```

Comment:
Just FYI. If you make async call, it won't throttle ingestion upfront because it will release ingest thread. That means, ingest request will get accepted but the document conversion will fail due to to many requests. Anyway, I guess it is how other embedding processor is implemented so I guess it won't be consistant if I ask to make a sync call here only.